app.ts

```
@@ -0,0 +1,330 @@
+
+let records = Math.floor((window.innerHeight - 160) / 40); // Estimate of available table space
+let trimStart: number;
+let trimEnd: number;
+let countRec: number;
+let isAppend: boolean;
+
+// Variables fetched once with load of window
+let RECORDCOUNT: number;
+let HEADERS: string[];
+
+// Global document elements
+let contentTable: HTMLElement | null = document.getElementById('content-table');
+let tableBody: HTMLTableSectionElement | null = contentTable!.querySelector("tbody");
+let tableHead: HTMLElement | null = document.getElementById("content-thead");
+let pageInfo: HTMLElement | null = document.getElementById('page-info');
+let firstBtn: HTMLElement | null = document.getElementById('first');
+let prevBtn: HTMLElement | null = document.getElementById('prev');
+let nextBtn: HTMLElement | null = document.getElementById('next');
+let lastBtn: HTMLElement | null = document.getElementById('last');
+let inputBox: HTMLElement | null = document.getElementById('id-search');
+
+
+// Fetch headers and record count
+async function getData() {
+	// API calls for record count and headers
+	RECORDCOUNT = await fetch('/recordCount')
+		.then(resp => resp.json());
+	HEADERS = await fetch('/columns')
+		.then(resp => resp.json());
+
+	// Populate table with fetched data
+	loadIntoTable(contentTable);
+}
+
+
+window.onload = () => {
+	trimStart = 0
+	trimEnd = trimStart + records - 1;
+
+	firstBtn?.setAttribute("disabled", "disabled");
+	prevBtn?.setAttribute("disabled", "disabled");
```

I feel that if these buttons don't exist then you should rather throw an error (Stop the app).